x/config/override.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2024 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+func newOverrideFromURL(configURL *url.URL) (func(string) (string, error), error) {
+	query := configURL.Opaque
+	values, err := url.ParseQuery(query)
+	if err != nil {
+		return nil, err
+	}
+	hostOverride, portOverride := "", ""
+	for key, values := range values {
+		switch strings.ToLower(key) {
+		case "host":
+			if len(values) != 1 {
+				return nil, fmt.Errorf("host option must has one value, found %v", len(values))
+			}
+			hostOverride = values[0]
+		case "port":
+			if len(values) != 1 {
+				return nil, fmt.Errorf("port option must has one value, found %v", len(values))
+			}
+			portOverride = values[0]
+		default:
+			return nil, fmt.Errorf("unsupported option %v", key)
+
+		}
+	}
+	return func(address string) (string, error) {
+		host, port, err := net.SplitHostPort(address)
+		if err != nil {
+			return "", fmt.Errorf("address is not valid host:port: %w", err)
```

Comment:
Instead of returning an error, we may also `return hostOverride + ":" + portOverride` here if `hostOverride != "" && portOverride != ""`.

Reply:
Done